src/components/calcite-tree-item/calcite-tree-item.tsx

```diff
@@ -50,16 +50,22 @@ export class CalciteTreeItem {
 
   @Watch("expanded")
   expandedHandler(newValue: boolean): void {
-    const items = getSlotted<HTMLCalciteTreeItemElement>(this.el, "children", {
-      all: true,
-      selector: "calcite-tree-item"
-    });
+    // handle newly appended items on parents
+    if (!newValue && this.parentTreeItem) {
+      const { expanded } = this.parentTreeItem;
+      this.parentExpanded = expanded;
+    } else {
+      const items = getSlotted<HTMLCalciteTreeItemElement>(this.el, "children", {
+        all: true,
+        selector: "calcite-tree-item"
+      });
 
-    items.forEach((item) => (item.parentExpanded = newValue));
+      items.forEach((item) => (item.parentExpanded = newValue));
+    }
   }
 
   /** @internal Is the parent of this item expanded? */
-  @Prop() parentExpanded = false;
+  @Prop({ mutable: true }) parentExpanded = false;
```

Comment:
Sidebar: Not for this PR, but I think it'd be cleaner to have each tree item observe their children and update this property directly instead of having children be aware of their parents.